x-pack/plugins/lens/public/indexpattern_datasource/operations/layer_helpers.ts

```diff
@@ -30,6 +30,81 @@ interface ColumnChange {
   shouldResetLabel?: boolean;
 }
 
+interface ColumnCopy {
+  layer: IndexPatternLayer;
+  columnId: string;
+  sourceColumn: IndexPatternColumn;
+  sourceColumnId: string;
+  shouldDeleteSource?: boolean;
+  indexPattern: IndexPattern;
+}
+
+export function copyColumn({
```

Comment:
Right now all references are *hidden columns*, but in the future we will have *references to visible columns*. We should add a TODO about this because I assume that we will not want to copy any visible columns.